src/azure-cli/azure/cli/command_modules/eventgrid/_params.py

```diff
@@ -204,7 +207,7 @@ def load_arguments(self, _):    # pylint: disable=too-many-statements
         c.argument('public_network_access', arg_type=public_network_access_type)
         c.argument('inbound_ip_rules', action=AddInboundIpRule, nargs='+')
         c.argument('sku', arg_type=sku_type)
-        c.argument('identity', arg_type=identity_type)
+        c.argument('identity', arg_type=identity_type, deprecate_info=c.deprecate(expiration='2.46.0'))
```

Comment:
This may cause unexpected CI failure in the future. We recommend removing a deprecated argument manually.

Reply:
Well. Another deprecation indeed caused problem today after upgrading Azure CLI version to 2.35.0: https://github.com/Azure/azure-cli/pull/20682#discussion_r840232833